dev/src/bundle.ts

```diff
@@ -142,7 +142,11 @@ export class BundleBuilder {
   private elementToLengthPrefixedBuffer(
     bundleElement: firestore.IBundleElement
   ): Buffer {
-    const buffer = Buffer.from(JSON.stringify(bundleElement), 'utf-8');
+    // Convert to a valid proto message object then take its json representation.
+    // This take cares of stuff like converting internal byte array fields
+    // to base64 encodings.
+    const message = BundleElement.fromObject(bundleElement).toJSON();
+    const buffer = Buffer.from(JSON.stringify(message), 'utf-8');
```

Comment:
Let's talk about this offline later, but this is quite an expensive operation. The steps are:

- Manually serialize documents in ProtobufJS format
- Read elements back Protobuf JS and convert to Proto3 JSON
- Parse Proto3 JSON into string

This seems like an okay "quick fix", but not the most elegant solution. We should figure out whether we are ok with this quick fix or whether we should directly produce Proto3 JSON. We could use a similar approach as the Web SDK and use different serialization options in our serializer depending on the target. While this might end up being a fair amount of work, it also helps us with the timestamp conversion.

Furthermore, can you verify whether the current conversion correctly translates numbers greater than 2^53 bits? You would have to turn on [useBigInt](https://github.com/googleapis/nodejs-firestore/blob/master/types/firestore.d.ts#L158) and then write a large integer value like such: `2147483647n`. I am asking because the JSON roundtrip might convert this type to a lossy floating point representation.

Reply:
Added a test for BigInt.